src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -477,14 +477,43 @@
     FrameClauseUnitGrammar=OneOf("ROWS", "RANGE"),
     # It's as a sequence to allow to parametrize that in Postgres dialect with LATERAL
     JoinKeywords=Sequence("JOIN"),
-    TableConstraintReferenceOptionGrammar=OneOf(
+    ReferentialActionGrammar=OneOf(
         "RESTRICT",
         "CASCADE",
         Sequence("SET", "NULL"),
         Sequence("NO", "ACTION"),
         Sequence("SET", "DEFAULT"),
     ),
     DropBehaviorGrammar=OneOf("RESTRICT", "CASCADE", optional=True),
+    ReferenceDefinitionGrammar=Sequence(
+        "REFERENCES",
+        Ref("TableReferenceSegment"),
+        # Foreign columns making up FOREIGN KEY constraint
+        Ref("BracketedColumnReferenceListGrammar", optional=True),
+        Sequence(
```

Comment:
Similar question as above -- wondering whether these optional parts (`MATCH ...` and `ON ...`) would be better as separate segments.

Reply:
Same thoughts as above here